File: Project Euler #3/C/solution3.c
#include <stdio.h>


int main(){
    int t; 
    scanf("%d",&t);
    while(t--){
        long n; 
        scanf("%ld",&n);
        long max_prime = 2;
        while(n%2 == 0){
            n /=2;
        }
        long i = 3;
        while(i*i <= n){
            while(n%i == 0){
                max_prime = i;
                n /=i;
            }
            i = i + 2;
        }
        if(max_prime > n)
            printf("%ld\n", max_prime);
        else
            printf("%ld\n", n);
            
            
    }
    return 0;
}

File: Project Euler #3/Python/solution3.py
if __name__ == "__main__":
    t = int(input().strip())
    for a0 in range(t):
        n = int(input().strip())
        max_prime = 2
        i = 3
        while n%2 == 0:
            n = n//2
            
        while i*i<=n:
            while n%i == 0:
                max_prime = i
                n = n//i
            i = i + 2
        if max_prime > n:    
            print(max_prime)
        else:
            print(n)    
